Raptor Add/Update buy event (#3141)
ezsystems/developer-documentation: docs/templating/twig_function_reference/recommendations_twig_functions.md

[[= include_file('code_samples/recommendations/events/content_visit_event.html.twig') =]]
```

### Product `buy` event

This event tracks when a product is bought.

- **Product object** defines the product being purchased.

> - **Context array with purchase conditions** - provide optional data about the product purchase context, like quantity, price, or currency.

Reviewer: ```suggestion
- **Context array with purchase conditions** - provides optional data about the product purchase context, like quantity, price, or currency.
```

[[= include_file('code_samples/recommendations/events/basket_event.html.twig') =]]
```

### `itemclicked` event

This event tracks when a user clicks a Raptor recommendation, including adding products to the cart from the recommendation module.

Required data:

> - **Product code** - defines the product code added to the cart.

Reviewer 2: ```suggestion
- **Product code** - product code of the product the visitor interacted with
```

I don't like my suggestion, feel it's clunky - but if I understand correctly, "clicks a Raptor recommendation" is broader than "added to cart" - can be "clicked and went to product page"? But then it's overlapping with product visit 🤔 

I know you're not even changing this, just moving 😓

Author: I would avoid saying product twice like in the following. I'm just wondering about the tense:

```suggestion
- **Product code** - code of the product the visitor interacted with.
```

```suggestion
- **Product code** - code of the product the visitor is interacting with.
```

Reviewer 2: Ok for me! I'd go for past tense, feel it's simpler to understand

{% endblock %}
```

> For another usage examples, see the [`buy`](#product-buy-event) and [`basket`](#product-basket-event) events.

Author: ```suggestion
For other usage examples, see the [`buy`](#product-buy-event) and [`basket`](#product-basket-event) events.
```
